backend/app.py

```diff
@@ -129,12 +132,53 @@ def fields_from_objlist(objtype: str = Query(...),
     queryClauses.insert(0, '.[]')
     query = ' | '.join(queryClauses)
     subset = jqQuery(objtype=objtype, objlist=objlist, query=query)
+
     return subset
     # groupQuery = 'group_by(.conceptSetNameOMOP) | map({ key: .[0].conceptSetNameOMOP | tostring, value: [.[] | {version, codesetId}] }) | from_entries'
     # res = jqQuery(objtype=objtype, objlist=subset, query=groupQuery)
     # return res
 
 
+
+@app.get("/concept-sets-with-concepts")
+def concept_sets_with_concepts(
+    codeset_id: Union[str, None] = Query(default=[]),
+    field: Union[List[str], None] = Query(default=[]),
+    concept_field_filter: Union[List[str], None] = Query(default=None),
+) -> Union[Dict, List]:
+    """Returns list of concept sets selected and their concepts
+
+    If no codeset_id, doesn't return concepts; just concept_sets."""
+    concept_sets = fields_from_objlist(objtype='OMOPConceptSet', filter=[f'codesetId:{codeset_id}'], field=field)
+    if not codeset_id:
+        return concept_sets
+    else:
+        # Mutate `concept_sets` by adding `concepts` field
+        concept_sets_lookup = {x['codesetId']: x for x in concept_sets}
+        concept_id_concepts_map = concepts_read(
+            concept_set_id=[x['codesetId'] for x in concept_sets], field_filter=concept_field_filter)
+        for cs_id, cs in concept_sets_lookup.items():
+            cs['concepts'] = concept_id_concepts_map[cs_id]
+
+        # TODO: Remove this block after we get all the data. This just fills in missing data for easy frontend rendering
+        for cs in concept_sets:
+            for concept_id, concept_props in cs['concepts'].items():
+                if not concept_props:
+                    cs['concepts'][concept_id] = {
+                        **{'conceptId': concept_id},
+                        **{f: '<Data not yet downloaded>' for f in concept_field_filter if f != 'conceptId'}}
+
+        return concept_sets
+
+
+# TODO:
+@app.get("/concept-sets-page")
+def concept_sets_page():
```

Comment:
FYI: I made this route to be called on "Cset search" page load, but I haven't implemented it yet. Might just remove it, given the recent ideas we came up with for loading all of the concepts when the backend process starts up.